mava/environment_loop.py

```diff
@@ -243,31 +243,41 @@ def should_run_loop(eval_interval_condition: Tuple) -> bool:
+            try:
+                if (not environment_loop_schedule) or (
+                    should_run_loop(eval_interval_condition)
+                ):
+                    if environment_loop_schedule:
+                        results = self.run_episode()
+                        # Get first result dictionary
+                        for _ in range(evaluation_duration - 1):
+                            # Add consecutive evaluation run data
+                            result = self.run_episode()
+                            # Sum results for computing mean after all evaluation runs.
+                            results = jax.tree_map(lambda x, y: x + y, results, result)
+                        # compute the mean over all evaluation runs
+                        results = jax.tree_map(
+                            lambda x: x / evaluation_duration, results
+                        )
+                        # Check for extra logs
+                        if hasattr(self._environment, "get_interval_stats"):
+                            results.update(self._environment.get_interval_stats())
+                        self._logger.write(results)
+                    else:
                         result = self.run_episode()
-                        # Sum results for computing mean after all evaluation runs.
-                        results = jax.tree_map(lambda x, y: x + y, results, result)
-                    # compute the mean over all evaluation runs
-                    results = jax.tree_map(lambda x: x / evaluation_duration, results)
-                    # Check for extra logs
-                    if hasattr(self._environment, "get_interval_stats"):
-                        results.update(self._environment.get_interval_stats())
-                    self._logger.write(results)
+                        # Log the given results.
+                        self._logger.write(result)
                 else:
-                    result = self.run_episode()
-                    # Log the given results.
-                    self._logger.write(result)
-            else:
-                # Note: We assume that the evaluator will be running less
-                # than once per second.
-                time.sleep(1)
-            # We need to get the latest counts if we are using eval intervals.
-            if environment_loop_schedule:
+                    # Note: We assume that the evaluator will be running less
+                    # than once per second.
+                    time.sleep(1)
+                # We need to get the latest counts if we are using eval intervals.
+                if environment_loop_schedule:
+                    self._executor.force_update()
+
+            except Exception as e:
+                print(e, ": Experiment terminated due to an error on a node.")
+                self._executor.store.executor_parameter_client.add_and_wait(
+                    {"interrupt": True}
```

Comment:
Thanks @OmaymaMahjoub :fire: Are we killing the same PID in the parameter server that we would have killed in the executor? It is still weird that this does not work. Maybe child processes cannot kill parent processes?

Reply:
Thank you @DriesSmit, yeah it is very strange it feels like the executor doesn't have any power on the parameter server that's why I send a flag to the param server and it will handle it itself

Reply:
Sounds good, thanks :slightly_smiling_face: